core/commonMain/src/ArgParser.kt

```diff
@@ -436,7 +436,9 @@ open class ArgParser(
     internal fun saveOptionWithoutParameter(argValue: ParsingValue<*, *>) {
         // Boolean flags.
         if (argValue.descriptor.fullName == "help") {
-            usedSubcommand?.saveOptionWithoutParameter(argValue)
+            usedSubcommand?.let {
+                parse(listOf("${it.optionFullFormPrefix}${argValue.descriptor.fullName}"))
+            }
```

Comment:
I don't think it's a good idea to call `parse` method recursively, considering that it's designed to be non-reentrant. Why do you need this call?

Reply:
Sorry, missed subcommand object when removed changes needed for debug.
Fixed. Now it isn't recursive, it just calls needed parsing for subcommand.

Reply:
Ok. So did I get it right that if a subcommand is specified before `--help`, then that `subcommand.parse` call does not return and prints help and exists by itself?

Reply:
Yes, because `parse` creates all lists with arguments and options for command.